File: SpeedReaderAndroid/Book.kt
package com.example.speedreader

import androidx.room.Entity
import androidx.room.PrimaryKey
import androidx.room.TypeConverter
import com.google.gson.Gson
import com.google.gson.reflect.TypeToken

@Entity
data class Book(
    @PrimaryKey(autoGenerate = true) val id: Int = 0,
    val title: String,
    val fullText: String,
    var lastReadPosition: Int = 0,
    val dateImported: Long = System.currentTimeMillis(),
    var bookmarks: List<Int> = emptyList()
)

class Converters {
    @TypeConverter
    fun fromString(value: String): List<Int> {
        val listType = object : TypeToken<List<Int>>() {}.type
        return Gson().fromJson(value, listType)
    }

    @TypeConverter
    fun fromList(list: List<Int>): String {
        return Gson().toJson(list)
    }
}

File: SpeedReader/Book.swift
import Foundation
import SwiftData

@Model
final class Book {
    var id: UUID
    var title: String
    var fullText: String
    var lastReadPosition: Int
    var dateImported: Date
    var bookmarks: [Int] // Array of word indices
    
    init(title: String, fullText: String) {
        self.id = UUID()
        self.title = title
        self.fullText = fullText
        self.lastReadPosition = 0
        self.dateImported = Date()
        self.bookmarks = []
    }
}
